Confus/ConfusServer/Player.cpp

```diff
@@ -199,8 +213,103 @@ namespace ConfusServer
+        
+        m_Connection->sendMessage(bitstreamOut, PacketReliability::UNRELIABLE);
+    }
+ 
+
+    void Player::setConnection(Networking::Connection* a_Connection)
+    {
+        m_Connection = a_Connection;
+
+        m_Connection->addFunctionToMap(static_cast<unsigned char>(Networking::EPacketType::Player), [this](RakNet::Packet* a_Data)
+        {
+            RakNet::BitStream bitstreamIn(a_Data->data, a_Data->length, false);
+
+            RakNet::RakNetGUID playerID;
+            bitstreamIn.IgnoreBytes(sizeof(unsigned char));
+            bitstreamIn.Read(playerID);
+            
+            // Only continue if the playerID in the packet matches ours.
+            if(playerID == static_cast<RakNet::RakNetGUID>(ID))
+            {
+                irr::core::vector3df position;
+                irr::core::vector3df rotation;
+                EPlayerState state;
+                RakNet::Time stateChangeTime;
+                int8_t playerHealth;
+                bool forwardKeyPressed;
+                bool backwardKeyPressed;
+                bool leftKeyPressed;
+                bool rightKeyPressed;
+
+                bitstreamIn.Read(position);
+                bitstreamIn.Read(rotation);
+                bitstreamIn.Read(state);
+                bitstreamIn.Read(stateChangeTime);
+                bitstreamIn.Read(playerHealth);
+                bitstreamIn.Read(forwardKeyPressed);
+                bitstreamIn.Read(backwardKeyPressed);
+                bitstreamIn.Read(leftKeyPressed);
+                bitstreamIn.Read(rightKeyPressed);
+                
+                setRotation(rotation);
+
+                if(state == EPlayerState::LightAttacking)
+                {
+                    startLightAttack();
+                }
+                else if(state == EPlayerState::HeavyAttacking)
+                {
+                    startHeavyAttack();
+                }
+
+                auto movementDirection = irr::core::vector3df();
+                if(forwardKeyPressed)
+                {
+                    movementDirection.Z = 1.0f;
+                }
+                else if(backwardKeyPressed)
+                {
+                    movementDirection.Z = -1.0f;
+                }
+                if(leftKeyPressed)
+                {
+                    movementDirection.X = -1.0f;
+                }
+                else if(rightKeyPressed)
+                {
+                    movementDirection.X = 1.0f;
+                }
+                //Rotate with the negative xz-rotation (around the Y axis), as
+                //the scene node convention seems to be clockwise while the rotate function
+                //is counter-clockwise
+                movementDirection.rotateXZBy(-CameraNode->getRotation().Y);
+                movementDirection = movementDirection.normalize();
+                //TODO: Make the player on the server have new physics as well.
```

Comment:
Though I appreciate the comment as a reminder for me (:P), you can remove it, I'll make sure this works.